### PR TITLE
Add support for watch @ resourceVersion, and improve tests.

### PR DESCRIPTION
**Pull request by brendandburns**
@drubin @cben

Builds on #40, will rebase after #40 merges.

### CODE REVIEW

**Comment by brendandburns** [comment]
@drubin @cben this is ready for review.

Thanks!

File: kubernetes/src/kubernetes/watch.rb [hunk 1]
```diff
@@ -23,9 +23,16 @@ def initialize(client)
...
     end
 
-    def connect(path, &_block)
+    def make_url(path, resource_version)
+      query = '?watch=true'
+      query += "&resourceVersion=#{resource_version}" if resource_version
+      path + query
+    end
+
+    def connect(path, resource_version, &_block)
```
Comment (drubin) [inline kubernetes/src/kubernetes/watch.rb]: Maybe its worth setting this as `resource_version = nil` so it isn't a breaking change?

File: examples/watch/watch.rb [hunk 2]
```diff
@@ -20,6 +20,6 @@
 
 watch = Kubernetes::Watch.new(client)
 
-watch.connect('/api/v1/namespaces') do |obj|
+watch.connect('/api/v1/namespaces', nil) do |obj|
```
Comment (drubin) [inline examples/watch/watch.rb]: I think it makes sense to annotate these examples if we have more than one param and especially if the value is nil

File: kubernetes/src/kubernetes/watch.rb (hunk 1 above)
Comment (brendandburns) [reply kubernetes/src/kubernetes/watch.rb]: done. (learn something new about Ruby every day :)

File: examples/watch/watch.rb (hunk 2 above)
Comment (brendandburns) [reply examples/watch/watch.rb]: this is fixed by the comment above....

**Comment by brendandburns** [comment]
comments addressed, please take another look.

**Pull request merged**